src/QuartermasterShaman.sol

```diff
@@ -0,0 +1,226 @@
+    uint256 length = _members.length;
+    uint256 delay = _calculateDelay();
+    uint256[] memory amounts = new uint256[](length);
+    address member;
+
+    for (uint256 i; i < length;) {
+      member = _members[i];
+      if (onboardingDelay[member] == 0 && SHARES_TOKEN.balanceOf(member) == 0) {
+        onboardingDelay[member] = delay;
+        amounts[i] = STARTING_SHARES(); // else 0
+      }
+
+      unchecked {
+        ++i;
+      }
+    }
+    emit OnboardedBatch(_members, amounts, delay);
+  }
+
+  /**
+   * @notice Offboards a batch of members from the DAO, if they are not wearing the member hat. Offboarded members
+   * lose their voting power, but keep a record of their previous shares in the form of loot.
+   * @param _members The addresses of the members to offboard.
+   */
+  function offboard(address[] calldata _members) external wearsCaptainHat(msg.sender) {
+    uint256 length = _members.length;
+    uint256 delay = _calculateDelay();
+    uint256[] memory amounts = new uint256[](length);
+    address member;
+    uint256 shares;
+
+    for (uint256 i; i < length;) {
+      member = _members[i];
+      shares = SHARES_TOKEN.balanceOf(member);
+      if (offboardingDelay[member] == 0 && shares > 0) {
+        offboardingDelay[member] = delay;
+        amounts[i] = shares; // else 0
+      }
+
+      unchecked {
+        ++i;
+      }
+    }
+
+    emit OffboardedBatch(_members, amounts, delay);
+  }
+
+  /**
+   * Executes onboarding
+   */
+  function quarter(address[] calldata _members) external {
+    uint256 length = _members.length;
+    uint256[] memory amounts = new uint256[](length);
+
+    for (uint256 i; i < length;) {
+      address member = _members[i];
+      if (onboardingDelay[member] != 0 && onboardingDelay[member] <= block.timestamp) {
+        delete onboardingDelay[member];
+        amounts[i] = STARTING_SHARES();
+      }
+
+      unchecked {
+        ++i;
+      }
+    }
+    BAAL().mintShares(_members, amounts);
+    emit Quartered(_members, amounts);
+  }
+
+  /**
+   * Executes onboarding
+   */
+  function unquarter(address[] calldata _members) external {
+    uint256 length = _members.length;
+    uint256[] memory amounts = new uint256[](length);
+
+    for (uint256 i; i < length;) {
+      address member = _members[i];
+      if (offboardingDelay[member] != 0 && offboardingDelay[member] <= block.timestamp) {
+        delete offboardingDelay[member];
+        amounts[i] = SHARES_TOKEN.balanceOf(member);
+      }
+
+      unchecked {
+        ++i;
+      }
+    }
+    BAAL().burnShares(_members, amounts);
+    emit Unquartered(_members, amounts);
+  }
+
+  /*//////////////////////////////////////////////////////////////
+                          PRIVATE FUNCTIONS
+  //////////////////////////////////////////////////////////////*/
+
+  /**
+   * @notice Adds votingPeriod x2 to the current time to allow for mutiny delay
+   */
+  function _calculateDelay() private view returns (uint256 delay) {
```

Comment:
If the Captain is a member of the DAO with enough votes to sponsor a proposal, they can accomplish the same thing as onboarding with a similar trust module but more directly by creating a proposal to add the member of their choice.

I would actually consider decreasing the delay to differentiate the onboarding pathway even more.

Reply:
Yeah, that's how I'm accomplishing it without this. My main goal is to give the captain hat unilateral control over DAO operations. So bringing crew aboard without votes is the intention here. I think the delay is needed to avoid sybil or collusion attacks by the captain.

Reply:
I guess my point is that giving a captain unilateral control presumably comes with some degree of trust anyways. Or another way to say it is that if you don't trust the captain enough to not require such a long delay, perhaps they shouldn't be the captain. But perhaps I'm misunderstanding your intention here.